third_party/rust/adblock/v0_8/crate/src/resources/resource_storage.rs

```diff
@@ -233,52 +316,167 @@ fn with_js_extension(scriptlet_name: &str) -> String {
     }
 }
 
-/// Parses the inner contents of a `+js(...)` block into a Vec of its comma-delimited elements.
-///
-/// A literal comma is produced by the '\,' pattern. Otherwise, all '\', '"', and ''' characters
-/// are erased in the resulting arguments.
-fn parse_scriptlet_args(args: &str) -> Vec<String> {
-    static ESCAPE_SCRIPTLET_ARG_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r#"[\\'"]"#).unwrap());
+/// Returns the index of the next unescaped separator, as well as a boolean indicating whether or
+/// not the string must be postprocessed to normalize any separators along the way.
+fn index_next_unescaped_separator(s: &str, separator: char) -> (Option<usize>, bool) {
+    assert!(separator != '\\');
```

Comment:
Would returning `None` (or a `Result`) be safer here, panic-wise?

Reply:
It's more of an internal invariant; `separator` is always passed as a hardcoded constant (or, directly from the hardcoded ones in the the `match` patterns).

I left it there mostly to keep the code self-documenting, although I'm not opposed to changing it to a `debug_assert!` instead.

Reply:
Ok, and it's a non-public function. I don't have a strong feeling about it, was just looking for crash vectors.

Thanks for writing the loop termination notes, btw.